names.lua

```diff
@@ -1,68 +1,118 @@
---rename items or units
+--rename items or units with the native interface
 --[====[
 
 names
 =====
+
 Rename units or items.  Usage:
+:-help:    print this help message
+:-if a first name is desired press f, leave blank to clear current first name
+:-if viewing an artifact you can rename it
+:-if viewing a unit you can rename them
 
-:-help:     print this help message
-:-item:     if viewing an item
-:-unit:     if viewing a unit
-:-first [Somename | "Some Names like This":
-            if a first name is desired, leave blank to clear current first name
 ]====]
+
+local gui = require 'gui'
+local dlg = require 'gui.dialogs'
+local widgets = require 'gui.widgets'
 local utils = require 'utils'
+local vw
+local trg
+local choices
 
 validArgs = validArgs or utils.invert({
  'help',
- 'item',
- 'unit',
- 'first'
 })
-
 local args = utils.processArgs({...}, validArgs)
-
 if args.help then
- print(
+    print(
 [[names.lua
 arguments:
     -help
         print this help message
-    -item
-    if viewing an item
-    -unit
-    if viewing a unit
-    -first Somename or "Some Names like This"
-    if a first name is desired, leave blank to clear current first name
+    if a first name is desired press f, leave blank to clear current first name
+    if viewing an artifact you can rename it
+    if viewing a unit you can rename them
 ]])
- return
+    return
 end
-
-if not df.viewscreen_layer_choose_language_namest:is_instance(dfhack.gui.getCurViewscreen()) then
-    choices = df.new(df.viewscreen_setupadventurest) ; choices.subscreen = 3 ; gui = require 'gui' ; gui.simulateInput(choices, 'A_RANDOM_NAME') ; gui.simulateInput(choices, 'A_CUST_NAME')
+namescr = defclass(namescr, gui.Screen)
+namescr.focus_path = 'names'
+function namescr:init()
+    if not dfhack.gui.getViewscreenByType(df.viewscreen_layer_choose_language_namest)==true then
```

Comment:
Instead of `if not x == true`, use `if not x`. What you have could be misinterpreted as `if not (x == true)`, but it's actually `if (not x) == true`. 

As far as I can tell, you're checking to see if the top viewscreen is *not* a `viewscreen_layer_choose_language_namest`. I'm not sure why you didn't use `is_instance` for this, but it would be more consistent. `getViewscreenByType` is useful if you want to get a viewscreen that's *somewhere* in the viewscreen stack by passing a second parameter.

Reply:
Weirdly I used the is_instance earlier and derped about the if not x logic.